frontend/lib/resolution/prims.cpp

```diff
@@ -583,7 +583,6 @@ CallResolutionResult resolvePrimCall(Context* context,
       type = primFieldByNum(context, ci);
       break;
 
-    case PRIM_CLASS_NAME_BY_ID:
     case PRIM_ITERATOR_RECORD_FIELD_VALUE_BY_FORMAL:
```

Comment:
I think that it might be useful to split all these cases out into a table or map that just maps the primitive to its type. Maybe we could consider doing this for all runtime primitives since they seem to have incredibly simple types.

The reason being that I find switch cases with a lot of label stacking to be really hard to read (eyes gloss over).

Reply:
I don't disagree, but I think we should put that off until we have more of these implemented and then do a whole refactor of this list. IOW, I don't think this PR is the right place to make that change.